dspace-api/src/main/java/org/dspace/content/CheckExistingUUIDGenerator.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.id.UUIDGenerator;
+
+/**
+ * Allows DSpaceObjects to provide a pre-determined UUID
+ *
+ * @author April Herron
+ */
+public class CheckExistingUUIDGenerator extends UUIDGenerator {
```

Comment:
I notice here there's no check to see if the UUID *is unique*.  Is the plan here that we are depending on the database layer to throw an error if someone attempts to create, e.g. two Items with the same UUID?   If so, I'd recommend we add tests to verify this will result in an error.  We obviously need UUIDs to be unique.

As a sidenote, we might want to consider renaming this to be `PredefinedUUIDGenerator`, so that it better aligns with `getPredefinedUUID()` method....the current name implies to me that is "checks" whether a UUID exists or not.

Reply:
Currently what will happen is it will just overwrite the other item with that UUID I added a test to reflect this process.